### PR TITLE
[#811] Case notification base code

### PR DESCRIPTION
**Pull request by Bartvaderkin**
Let's review this part before I gets out of hand:

This is everything to accept notifications from Openzaak, including filtering the case/status/role/users, up to actually sending the message.

So these are basically two parts: the generic ZGW notification webhook and the first of the OIP specific part, but the PR touches a bunch of unrelated things because I updated the TimelineLog system-action wrapper.

I have a branch on top of this that adds the ZaakTypeConfig (to also handle eSuite's' informeren) and sending the actual email.

(not a lot of this would run locally so if the tests look good it's good)

### CODE REVIEW

File: src/open_inwoner/openzaak/api_models.py [hunk 1]
```diff
@@ -137,10 +137,52 @@ class Resultaat(ZGWModel):
...
 @dataclass
 class Status(ZGWModel):
     url: str
     zaak: Union[str, Zaak]
     statustype: Union[str, StatusType]
     datum_status_gezet: Optional[datetime] = None
     statustoelichting: Optional[str] = ""
+
+
+@dataclasses.dataclass
```
Comment (vaszig) [inline src/open_inwoner/openzaak/api_models.py:162]: Is it different from the one used above?It seems to be the same import.

File: src/open_inwoner/openzaak/auth.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                    "verify_iat": False,
+                    "require": ["client_id"],
+                },
+            )
+        except InvalidTokenError:
+            continue
+        else:
+            return subscription
+
+    raise InvalidAuthForClientID(client_id)
```
Comment (vaszig) [inline src/open_inwoner/openzaak/auth.py:67]: I don't get when this line should be hit.

File: src/open_inwoner/openzaak/models.py [hunk 3]
```diff
@@ -1,4 +1,6 @@
 from django.db import models
+from django.db.models import UniqueConstraint
+from django.utils import timezone
```
Comment (vaszig) [inline src/open_inwoner/openzaak/models.py:3]: We don't use these somewhere as I can see.

File: src/open_inwoner/openzaak/api/urls.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+from open_inwoner.openzaak.api.views import ZakenNotificationsWebhookView
+
+app_name = "openzaak"
+
+urlpatterns = [
+    path(
+        "notifications/webhook/zaken",
+        ZakenNotificationsWebhookView.as_view(),
+        name="notifications-webhook-zaken",
+    ),
```
Comment (vaszig) [inline src/open_inwoner/openzaak/api/urls.py:12]: A minor detail..if we want be consistent with the naming conventions of the project we use `notifications_webhook_zaken`

File: src/open_inwoner/urls.py [hunk 5]
```diff
@@ -76,6 +76,10 @@
     path("accounts/", include("django_registration.backends.one_step.urls")),
     path("accounts/", include("django.contrib.auth.urls")),
     path("api/", include("open_inwoner.api.urls", namespace="api")),
+    path(
+        "api/openzaak/",
+        include("open_inwoner.openzaak.api.urls", namespace="openzaak-api"),
```
Comment (vaszig) [inline src/open_inwoner/urls.py:81]: Same as above, naming with underscores.

File: src/open_inwoner/openzaak/api_models.py (hunk 1 above)
Comment (Bartvaderkin) [reply src/open_inwoner/openzaak/api_models.py:162]: Not really, just sloppy use of auto-imports. Fixed it.

File: src/open_inwoner/openzaak/auth.py (hunk 2 above)
Comment (Bartvaderkin) [reply src/open_inwoner/openzaak/auth.py:67]: This happens when we do have a Subscription(s) for the `client_id` we got from the unverified JWT, but the secret(s) is wrong (eg: it didn't hit the return in the loop above).

File: src/open_inwoner/openzaak/api/urls.py (hunk 4 above)
Comment (Bartvaderkin) [reply src/open_inwoner/openzaak/api/urls.py:12]: Yes we want.

**Review by vaszig — APPROVED** [review]

**Comment by alextreme** [comment]
Okay, lets give this a try on the testenvironment due to the dependancy on OpenZaak/OpenNotificaties

**Pull request merged**